data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/AvroConsumer.java

```diff
@@ -83,6 +90,30 @@ public void consumeRecords(final KafkaConsumer<String, GenericRecord> consumer,
         }
     }
 
+    private void iterateRecordPartitions(KafkaSourceBufferAccumulator kafkaSourceBufferAccumulator,
```

Comment:
These 3 functions are repeated in all consumers. is it not possible to abstract them out in super class? or have one Consumer class and inject deserializers based on schema avro/json/plaintext